internal/functions/checkattemptsthresholds/check-attempts-thresholds.go

```diff
@@ -0,0 +1,119 @@
+package checkattemptsthresholds
+
+import (
+	"context"
+	"encoding/json"
+	ers "errors"
+	"fmt"
+	"github.com/covid19cz/erouska-backend/internal/constants"
+	"github.com/covid19cz/erouska-backend/internal/logging"
+	"github.com/covid19cz/erouska-backend/internal/store"
+	"github.com/covid19cz/erouska-backend/internal/utils"
+	"github.com/covid19cz/erouska-backend/internal/utils/errors"
+	httputils "github.com/covid19cz/erouska-backend/internal/utils/http"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"net/http"
+	"strconv"
+)
+
+type queryRequest struct {
+	Ehrid string `json:"ehrid" validate:"required"`
+	IP    string `json:"ip" validate:"required"`
+	Date  int    `json:"date"`
+}
+
+type queryResponse struct {
+	ThresholdOk bool `json:"thresholdsOK"`
+}
+
+//CheckAttemptsThresholds Check if attempts are over threshold
+func CheckAttemptsThresholds(w http.ResponseWriter, r *http.Request) {
+	var ctx = r.Context()
+	logger := logging.FromContext(ctx)
+	client := store.Client{}
+
+	var request queryRequest
+
+	err := httputils.DecodeJSONBody(w, r, &request)
+	if err != nil {
+		var mr *errors.MalformedRequestError
+		if ers.As(err, &mr) {
+			logger.Debugf("Cannot handle CheckAttemptsThresholds request: %+v", mr.Msg)
+			logger.Error(err)
+			http.Error(w, mr.Msg, mr.Status)
+		} else {
+			logger.Debugf("Cannot handle CheckAttemptsThresholds request due to unknown error: %+v", err.Error())
+			logger.Error(err)
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		}
+		return
+	}
+
+	logger.Debugf("Handling CheckAttemptsThresholds request: %+v", request)
+
+	var date string
+	if request.Date == 0 {
+		date = utils.GetTimeNow().Format("20060102")
+	} else {
+		date = strconv.Itoa(request.Date)
+	}
```

Comment:
can we hide this in some internal/utils function?

Reply:
Good point. Will solve in following PR.

Reply:
Well... maybe not in this case :smile: I'm not actually sure how to do that - let'S discuss in person.